Packs/CommonScripts/Scripts/GetServerURL/GetServerURL.js

```diff
@@ -0,0 +1,34 @@
+  function parseUrl(url) {
+      pattern = /^(https?:)\/\/([^:/?#]+)(?::(\d+))?([^?#]*)(\?[^#]*)?(#.*)?$/;
+      match = url.match(pattern);
+
+      return match ? {
+          shcema: url.match(/^(https?):/)[1],
+          hostname: match[2],
+          port: match[3] || '',
+      } : null;
```

Comment:
Where did you get the regex from?
Please consult with @Ni-Knight and consider adding to CommonServer.js